### ISSUE TITLE
kapp creates state configmap on --diff-run

### ISSUE THREAD

**Issue by schmurfy** [k14s/kapp#432]
**What steps did you take:**
kapp deploy -a myapp -f . --diff-run

**What happened:**
kapp shows the diff but created an empty configmap for this app, when running `kapp list` afterwards the app is shown in the list with "-" as namespace, an inspect list nothing, as it should

**What did you expect:**
nothing changed on the cluster

**Anything else you would like to add:**
[Additional information that will assist in solving the issue.]

**Environment:**

- kapp version (use `kapp --version`): kapp version 0.42.0 (I checked but no changes on this appear on the changelogs for the following releases)
- OS (e.g. from `/etc/os-release`): osx, linux
- Kubernetes version (use `kubectl version`)
```
lient Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T18:49:28Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.12-gke.1500", GitCommit:"d32c0db9a3ccd0ac73b0b3abd0532505217b376e", GitTreeState:"clean", BuildDate:"2021-11-17T09:30:02Z", GoVersion:"go1.15.15b5", Compiler:"gc", Platform:"linux/amd64"}
```

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by 100mik** [comment]
Looks like this issue is also being tracked in this one #23

**Comment by rohitagg2020** [comment]
@rohitagg2020 Please check: When `diff-run` is performed against the labeled app, we don't create configmap. Can we convert recorded app to labeled app and do the `diff-run`.

**Comment by schmurfy** [comment]
@rohitagg2020 What are labeled apps and recorded apps ?

**Comment by cppforlife** [comment]
rohit is referring to using `-a label:x=y` (labeled) instead of `-a app-name` (recorded). currently only recorded apps get "created" even for --diff-run. (*note: i always recommend using recorded apps in 99% cases. labeled apps were introduced as an advanced/specialized option, especially useful for browsing cluster e.g. `kapp inspect -a label:`)

i think your ask is totally reasonable and we should address it, since if the app doesnt exist its only fair it should not get created with --diff-run (current behaviour was intentionally left for implementation simplicity, so not a bug, but it's probably time to improve it).

**Comment by rohitagg2020** [comment]
@schmurfy I apologize for being ambiguous. It was a note to self.

**Comment by renuy** [comment]
fix available as part of v0.48.0
